StrongInject.Generator/DisposalLowerer.cs

```diff
@@ -45,6 +45,8 @@ source switch
                             } 
                             : null,
                         DelegateParameter or InstanceFieldOrProperty or ArraySource or ForwardedInstanceSource => null,
+                        // TODO: write test injecting Owned<T> directly to see whether this is the right thing to do:
+                        OwnedSource owned => ExactTypeKnown(owned.ContainerType, variableToDisposeName),
```

Comment:
I don't think we want to dispose this at all, since it's up to the user to dispose this. Holding on to it for disposal would cause memory leaks.

Reply:
Interesting. I would have expected StrongInject to dispose an `Owned<T>` instance for me when it disposes the last thing using it, the same way it disposes any other injected IDisposable instance for me when it disposes the last thing using it. Does `Owned<T>` have to be special-cased in this way?

Reply:
Now that I'm thinking longer about this, I would not expect the container to dispose `Owned<T>` instances I got by invoking an injected `Func<Owned<T>>`, because I might need control over when the thing is actually disposed and it might be some time later than the component that invoked the func, after an async wait perhaps. So I guess I would expect the same to apply to instances injected into `Owned<T>` parameters too: the whole point of doing so would be to have full manual control over the disposal of the `Owned<T>` instance, and therefore StrongInject should never dispose it for you.

Reply:
There's one thing that makes me think it would be okay for StrongInject to forcibly dispose injected `Owned<T>` instances and all `Owned<T>` instances obtained by an injected `Func<Owned<T>>`: if StrongInject waits to do this until the component that requested these instances completes its IDisposable.Dispose() implementation or its IAsyncDisposable.DisposeAsync() implementation. Since async disposal is available in StrongInject, the argument could be made that the component should not allow its own disposal to complete until it is ready to release all `Owned<T>` instances it was ever given.

The reason to do this would be to avoid leaks if the component forgets to dispose the `Owned<T>` instances. It would be extra complexity to track these instances at runtime though, and StrongInject could take the position that the safety is off at runtime and you should just know better than to forget to dispose these instances.

Reply:
A great analysis. I think that I would agree with the last position you took - not disposing of an `Owned` is a critical error, and StrongInject offers you no assistance in such cases.

Reply:
In the meantime, safety can be added later on if you decided that 1) there is no good reason for a component to need access to `Owned<T>` values after the component's Dispose/DisposeAsync completes and 2) it was worth the effort and overhead. For now, I'd rather remove the TODO comment and make this a no-op.

Reply:
Sounds good!